### ISSUE TITLE
why advantage calculate  ops [::-1]

### ISSUE THREAD

**Issue by DavideHe** [OpenRLHF/OpenRLHF#437]
as the [code](https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/trainer/ppo_utils/experience_maker.py#L225) ,`advantages = torch.stack(advantages_reversed[::-1], dim=1)` . 
why  the code operate this `advantages_reversed[::-1]` ?  if the right reponse pad <pad_ids> , how does the advantage align the valid  response token ?
